File: 1. Merge Strings Alternately/MergeStringsAlternately.go
func mergeAlternately(word1 string, word2 string) string {
    word1_slice := strings.Split(word1, "")
    word2_slice := strings.Split(word2, "")
    merged := ""
    i := 0

    for (len(merged) < len(word1) + len(word2)) {
        if (len(word1_slice) > i) {
            merged += word1_slice[i]
        }

        if (len(word2_slice) > i) {
            merged += word2_slice[i]
        }

        i++
    }

    return merged
}

File: 1. Merge Strings Alternately/MergeStringsAlternately.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: str
        """
        word1_list = list(word1)
        word2_list = list(word2)
        merged = ""
        i = 0

        while i < len(word1) + len(word2):
            if len(word1_list) > 0:
                merged += word1_list.pop(0)

            if len(word2_list) > 0:
                merged += word2_list.pop(0)
            
            i += 1
        
        return merged